### ISSUE TITLE
How to display a movie? / missing pylab.cla or clf

### ISSUE THREAD

**Issue by jeammimi** [maartenbreddels/ipyvolume#4]
Hi!
Your library seems quite interesting and I would like to use it to render some molecular dynamic
movie.
But for this I would need to be able to add the time dimension. I am not very familiar with ipywidget, and I tried  to do it with interact:
```python
from IPython.html.widgets import interact, fixed,interact_manual,IntSlider,VBox
import numpy as np
import ipyvolume
import ipyvolume.pylab as p3
import numpy as np

p3.figure()

def toi(time): 
    x = np.array([ 2 * t + np.arange(10) for t in range(5)])
    y = x**2
    z = 3 * x
    #x,y,z are two dimentionnal array with the first dimension being time
    s = p3.scatter(x[time],y[time],z[time], s=0.005)
    

interact(toi,time=IntSlider(min=0,max=4,step=1,value=0))

VBox([p3.gcc()])
```

But it does not really do what I want, because it does not remove the old points when I move through time. Also I think it does some rescaling.
Do you have any sugestion?

Thanks!

**Comment by maartenbreddels** [comment]
Hi,

thanks for the report. Love to see this kind of feedback to see how people use it :+1:.

I don't think this is a bug, calling scatter multiple times for the same figure (as in matplotlib) should make multiple scatters. The rescaling is because if you call scatter, it will set the limits such that it fits the 'box'.
However, it should be possible to clear up the current figure, which is not yet possible.
A workaround would be this:
```python
from IPython.html.widgets import interact, fixed,interact_manual,IntSlider,VBox
import numpy as np
import ipyvolume
import ipyvolume.pylab as p3
import numpy as np

f = p3.figure()
f.animation = 100

#x,y,z are two dimentionnal array with the first dimension being time
x = np.array([ 2 * t + np.arange(10) for t in range(5)])
y = x**2
z = 3 * x

p3.xlim(0, np.max(x))
p3.ylim(0, np.max(y))
p3.zlim(0, np.max(z))

time = 0
s = p3.scatter(x[time],y[time],z[time], s=0.005)

def toi(time): 
    s.x = x[time]
    s.y = y[time]
    s.z = z[time]
    

interact(toi,time=IntSlider(min=0,max=4,step=1,value=0))

VBox([p3.gcc()])
```


If you don't like the animation between the positions, or the speed of it:
```python
f = p3.gcf()
f.animation = 0
```
where the animation time is on msec, 100 seems to look nice (currently undocumented, and subject to change).

Note that from anywhere in the notebook you can do
```python
s.x = x[0]**0.5
```
And it will update it.


I'll keep this issue open until i have a matplotlib `pylab.cla()` or `pylab.clf()` equivalent.

**Comment by maartenbreddels** [comment]
If you have something nice to show, let me know. It may serve as a good example.

**Comment by jeammimi** [comment]
Ok, I will

**Comment by jeammimi** [comment]
Do you think it is possible to embed the slider and the ipvolume rendering in an html
I tried this,

```python
time= IntSlider(min=0,max=4,step=1,value=0)
it = interact(toi,time=time)

vbox = VBox([p3.gcc(),time])
ipyvolume.embed.embed_html("test.html",vbox)
```

But the slider does nothing

**Comment by maartenbreddels** [comment]
No, since this requires Python code running. Only when you do jslink (like in the example on the homepage of https://ipyvolume.readthedocs.io ) it works. I am thinking about how to solve this btw, not sure when it will be solved and how, but it is something that I want to support.

**Comment by jeammimi** [comment]
Ok, So to do that, I would have for example to do myself a new ipyvolume scatter function that would take x,y,z with two dimension and the time as an argument.

**Comment by maartenbreddels** [comment]
I am thinking of doing exactly that, and then adding an `.index` property to the scatter such that you could jslink `s.index` to the slider.

**Comment by jeammimi** [comment]
Ok, but I don't want to give you to much work.
If you want I can try to do it.

**Comment by maartenbreddels** [comment]
It's not that trivial, it requires changes on the javascript side as well. Feel free to try this out and submit a pull request.

**Comment by jeammimi** [comment]
I was thinking indeed that it was not trivial.
Do you think that it is possible to try to write a class that inheritate your scatter class and add the index at this level or do you think it is needed to add the index to the scatter class itself.
I think the first case would be better, because it would be more flexible, but I am not sure if it is possible.

**Comment by maartenbreddels** [comment]
I don't mind having an extra `.index` for Scatter. Most work would would go into the ScatterView in [volume.js](https://github.com/maartenbreddels/ipyvolume/blob/master/js/src/volume.js#L188). It would need to recognize if x y and z are 2d instead of 1d, and if so, use the `this.model.get('index')` value as an index for that. As well, it requires listening to changes to this property (similar as it listens to changes in x y, z and color), and making sure the animation goes well. Not so much work, I'd say.

**Comment by jeammimi** [comment]
Ok, then I will try to do it

**Comment by maartenbreddels** [comment]
I use this for development in the notebook, to avoid refreshing the notebook:
https://github.com/maartenbreddels/ipyvolume/blob/master/examples/dev.ipynb
But you can also run the embed function so you can simply refresh a static html  page instead of the notebook. In this case, make sure that if you save it to say `/tmp/test.html` that the generated file in ipyvolume/ipyvolume/static/index.js is copied (or better symlinked) to `/tmp/ipyvolume.js`
And check this: https://github.com/maartenbreddels/ipyvolume#installation 
Running the webpack --watch is handy.

**Comment by jeammimi** [comment]
Hello,
I did it and it was not too difficult. It works in the notebook but not in the embedded version.
Is it because in the embedded it look for the javascript in another place ?

**Comment by maartenbreddels** [comment]
Sound good! Yes, as I said in the msg above here. If you open a file from <path> in the browser, it will look for <path>/ipyvolume.js (see the chrome dev console for that). If it cannot find that, it will get it online from a cdn (you can also see that in the dev console).
Try it out, and hope to see a PR soon!

**Comment by jeammimi** [comment]
I talked a little bit fast.
I managed to make it work and to record the correct previous positions,
I also added animation_time_index : { type: "f", value: 1. },
But it does no transition. It just put the correct new coordinates
What did I forget ?

**Comment by maartenbreddels** [comment]
I don't think you need that extra variable in the shader, you can simply use the animation_time_{x,y,z}. Why don't you do a PR, so I can take a look at your code?

**Comment by jeammimi** [comment]
Ok, I just did it

**Comment by maartenbreddels** [comment]
:+1: hope I can comment tomorrow on it!

**Comment by maartenbreddels** [comment]
Fixed by #6
